py-polars/polars/_utils/construction/series.py

```diff
@@ -500,6 +499,8 @@ def numpy_to_pyseries(
         return constructor(
             name, values, nan_to_null if dtype in (np.float32, np.float64) else strict
         )
+    # TODO: remove this branch on 1.0.
+    # This returns a List whereas we should return an Array type
     elif values.ndim == 2:
```

Comment:
This branch still returns `List<i64>` for 2D arrays. Will remove that during 1.0, so we return an `Array` type here.